temporian/core/operators/cast.py

```diff
@@ -0,0 +1,259 @@
+from temporian.proto import core_pb2 as pb
+
+
+class CastOperator(Operator):
+    """Type cast operator."""
+
+    def __init__(
+        self,
+        event: Event,
+        to_dtype: Optional[DType] = None,
+        from_dtypes: Optional[Mapping[DType, DType]] = None,
+        from_features: Optional[Mapping[str, DType]] = None,
+        check_overflow: bool = True,
+    ):
+        super().__init__()
+
+        # Check that provided arguments are coherent
+        self._check_args(event, to_dtype, from_dtypes, from_features)
+
+        # Convert to {feature_name: target_dtype}
+        target_dtypes = self._get_feature_dtype_map(
+            event, to_dtype, from_dtypes, from_features
+        )
+
+        # Attributes
+        self.add_attribute("target_dtypes", target_dtypes)
+        self.add_attribute("check_overflow", check_overflow)
+
+        # Inputs
+        self.add_input("event", event)
+
+        # Output event features
+        output_features = []
+        reuse_event = True
+        for feature in event.features:
+            if target_dtypes[feature.name] is feature.dtype:
+                # Reuse feature
+                output_features.append(feature)
+            else:
+                # Create new feature
+                reuse_event = False
+                output_features.append(
+                    # Note: we're not renaming output features here
+                    Feature(
+                        feature.name,
+                        target_dtypes[feature.name],
+                        feature.sampling,
+                        creator=self,
+                    )
+                )
+
+        # Output event: don't create new if all features are reused
+        self.add_output(
+            "event",
+            event
+            if reuse_event
+            else Event(
+                features=output_features,
+                sampling=event.sampling,
+                creator=self,
+            ),
+        )
+
+        self.check()
+
+    def _check_args(
+        self,
+        event: Event,
+        to_dtype: Optional[DType] = None,
+        from_dtypes: Optional[Mapping[DType, DType]] = None,
+        from_features: Optional[Mapping[str, DType]] = None,
+    ) -> None:
+        # Check that only one of these args was provided
+        oneof_args = [to_dtype, from_dtypes, from_features]
+        if sum(arg is not None for arg in oneof_args) != 1:
+            raise ValueError(
+                "One and only one of to_dtype, from_dtypes or from_features"
+                " should be provided to CastOperator."
+            )
+
+        # Check: to_dtype is actually a dtype
+        if to_dtype is not None and not isinstance(to_dtype, DType):
+            raise ValueError(f"Cast: expected DType but got {type(to_dtype)=}")
+
+        # Check: from_dtypes is a dict {dtype: dtype}
+        if from_dtypes is not None and (
+            not isinstance(from_dtypes, Mapping)
```

Comment:
does `isinstance()` work with Mapping?

I know it doesn't work with something like this:

```python
if isinstance(x, Union[int, float])
```

Reply:
Yes 👍 
```
>>> a = {"hello": 1}
>>> isinstance(a, Mapping)
True
```